doc/source/myrocks/data_loading.rst

```diff
@@ -0,0 +1,157 @@
+  operations, and then returning and bulk loading addition data if there is no
+  overlap between what is being loaded and what already exists.
+
+* The data may not be visible until bulk load mode is ended (i.e. the
+  :variable:`rocksdb_bulk_load` is set to zero again). The method that is used
+  is building up SST files which will later be added as-is to the database.
+  Until a particular SST has been added the data will not be visible to the
+  rest of the system, thus issuing a ``SELECT`` on the table currently being
+  bulk loaded will only show older data and will likely not show the most
+  recently added rows. Ending the bulk load mode will cause the most recent SST
+  file to be added. When bulk loading multiple tables, starting a new table
+  will trigger the code to add the most recent SST file to the system -- as a
+  result, it is inadvisable to interleave ``INSERT`` statements to two or more
+  tables during bulk load mode.
+
+By default, the :variable:`rocksdb_bulk_load` mode expects all data be inserted
+in primary key order (or reversed order). If the data is in the reverse order
+(i.e. the data is descending on a normally ordered primary key or is ascending
+on a reverse ordered primary key), the rows are cached in chunks to switch the
+order to match the expected order.
+
+Inserting one or more rows out of order will result in an error and may result
+in some of the data being inserted in the table and some not. To resolve the
+problem, one can either fix the data order of the insert, truncate the table,
+and restart.
+
+Unsorted bulk loading
+=====================
+
+If your data is not ordered in primary key order, then this method is
+recommended. With this method, secondary keys do not need to be dropped and
+restored. However, writing to the primary key no longer goes directly to SST
+files, and are written to temporary files for sorted first, so there is extra
+cost to this method.
+
+To allow for loading unsorted data:
+
+.. code-block:: mysql
+
+  SET session sql_log_bin=0;
+  SET session rocksdb_bulk_load_allow_unsorted=1;
+  SET session rocksdb_bulk_load=1;
+  ...
+  SET session rocksdb_bulk_load=0;
+  SET session rocksdb_bulk_load_allow_unsorted=0;
+
+Note that :variable:`rocksdb_bulk_load_allow_unsorted` can only be changed when
+:variable:`rocksdb_bulk_load` is disabled (set to ``0``). In this case, all
+input data will go through an intermediate step that writes the rows to
+temporary SST files, sorts them rows in the primary key order, and then writes
+to final SST files in the correct order.
+
+Other Approaches
+================
+
+If :variable:`rocksdb_bulk_load` does not fit your needs there are two other
+settings that might help: :variable:`unique_check` and
+:variable:`rocksdb_commit_in_the_middle`. By default :variable:`unique_check`
+is enabled and :variable:`rocksdb_commit_in_the_middle` is disabled.
+
+If :variable:`unique_check` is disabled, MyRocks skips unique constraint
```

Comment:
'unique_check' is actually currently 'broken' in PS in that we ignore the request to skip unique checks on PK as it can/will cause index corruption in MyRocks if the keys are not structured properly. We 'fixed' this in TokuDB by validating key structure before allowing the 'blind' operations and if the key structure would cause/allow inconsistencies, we do the unique check anyway.

So for the purpose of this documentation, I would not mention the unique_check option and maybe make a not that we currently differ from upstream Facebook here.